### ISSUE TITLE
Enhancement: Configurable min, max, and default widths for resizable panel

### ISSUE THREAD

**Issue by bpatterson88** [Esri/calcite-components#3615]
### Description

Allow for configurable min, max, and default widths for a resizable panel. 

### Acceptance Criteria 

- Min width is available as a property for resizable panels
- Max width is available as a property for resizable panels
- Default width is available as a property for resizable panels

#### Which Component

`shell-panel`

#### Example Use Case

Dashboards places a calcite-tree of unpredictable width inside panels. When users attempt to resize the panel, they should (99% of the time) be able to resize the panel so that all labels are readable. Allowing the developer to set the maximum width of the panel would make this possible.

![128782796-d48e4934-1058-48b7-9296-fb1a6385295e](https://user-images.githubusercontent.com/15875886/144331476-f4bf3055-be66-4a09-85fc-dac9ac022354.png)

**Comment by macandcheese** [comment]
These values actually already exist: https://github.com/Esri/calcite-components/blob/master/src/components/calcite-shell-panel/calcite-shell-panel.scss#L84-L86
```
 width: var(--calcite-shell-panel-width);
 max-width: var(--calcite-shell-panel-max-width);
 min-width: var(--calcite-shell-panel-min-width);
```


but they aren't documented and there's currently an issue preventing them from working if set externally because they are set again here: https://github.com/Esri/calcite-components/blob/master/src/components/calcite-shell-panel/calcite-shell-panel.scss#L105-L121

Since `width-scale` defaults to `m` when not set, these are *always* get overridden.

For now there's a workaround by setting a "fake" value : `<calcite-shell-panel width-scale="not-a-real-scale ... >`, which will allow custom set variables to be interpreted: 
https://jsbin.com/samiqocito/1/edit?html,js,output


Thoughts on this @asangma - I think we should probably let these values be set by a user all the time. It might also let us simplify customizing width of panel and shell-panel and look at how `width-multiplier` is used.

**Comment by macandcheese** [comment]
These values actually already exist: https://github.com/Esri/calcite-components/blob/master/src/components/calcite-shell-panel/calcite-shell-panel.scss#L84-L86
```
 width: var(--calcite-shell-panel-width);
 max-width: var(--calcite-shell-panel-max-width);
 min-width: var(--calcite-shell-panel-min-width);
```


but they aren't documented and there's currently an issue preventing them from working if set externally because they are set again here: https://github.com/Esri/calcite-components/blob/master/src/components/calcite-shell-panel/calcite-shell-panel.scss#L105-L121

Since `width-scale` defaults to `m` when not set, these *always* get overridden.

For now there's a workaround by setting an invalid value : `<calcite-shell-panel width-scale="" ... >`, which will allow custom set variables to be interpreted: 
https://jsbin.com/nicogopebu/1/edit?html,js,output


Thoughts on this @asangma - I think we should probably let these values be set by a user all the time. It might also let us simplify customizing width of panel and shell-panel and look at how `width-multiplier` is used.

**Comment by MichalisM** [comment]
Yes, please! Can you make them publicly available? It's going to be super helpful!

**Comment by driskull** [comment]
> Since width-scale defaults to m when not set, these always get overridden.

Maybe we shouldn't do this part either.
